llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -2151,15 +2151,15 @@ template <typename Container, typename UnaryFunctor, typename NullaryFunctor,
               !std::is_constructible<StringRef, NullaryFunctor>::value>>
 inline void interleave(const Container &c, UnaryFunctor each_fn,
                        NullaryFunctor between_fn) {
-  interleave(c.begin(), c.end(), each_fn, between_fn);
+  interleave(adl_begin(c), adl_end(c), each_fn, between_fn);
```

Comment:
Curiosity since I was looking at making the same change: What's the difference between using `std::begin` vs `adl_begin` here (especially when compared to the existing `.begin()`).

Reply:
So in the case of upgrading from `.begin()` it's not to keep compatibility with anything, but to make this behave more nicely for future types. Thanks for confirming!

Reply:
We should do a clean up pass and use adl_begin/end across the whole STLExtrass. There's no reason to use the `std::` version anywhere there.

Reply:
Agreed, but I don't want to do that as part of this patch, so as to quickly unblock https://github.com/llvm/llvm-project/pull/87585 doing the "right thing".

Reply:
I've refactored that other PR slightly and I think the refactor makes `ArrayRef<>` the most appropriate type there, so this is no longer blocking that.

Reply:
> Agreed, but I don't want to do that as part of this patch, so as to quickly unblock https://github.com/llvm/llvm-project/pull/87585 doing the "right thing".

yep, absolutely